pddl/custom_types.py

```diff
@@ -43,3 +43,17 @@ class name(RegexConstrainedString):
 def to_names(names: Collection[namelike]) -> List[name]:
     """From name-like sequence to list of names."""
     return list(map(name, names))
+
+
+def to_names_types(
+    names: Dict[namelike, Collection[namelike]]
+) -> Dict[name, Collection[name]]:
+    """From name-like collection to name dictionary."""
+    return (
+        {
+            name(type_def): [name(list(parents)[0])]
+            for type_def, parents in names.items()
+        }
+        if isinstance(names.values(), list)
+        else {name(type_def): [] for type_def in names}
+    )
```

Comment:
This block's a little inscrutable for me. Can you rewrite with some commentary as to what's going on here? Both the indexing of `[0]` and `isinstance(names.values(), list)` look a little sus ;).

Reply:
When parsing `(:types ...`, the type structure has the following types Dict[Token, Set]. Now, every type can have a single parent, so the `[0]` is to extract that element. (Note here we do not allow `(either ...`. 

When parsing `(:predicates ...`, each parameter may have >1 types with `either`, so a list as dict value is needed here.

Finally, the `[]` is to allow domains with types without hierarchy.